test/reinit_test.jl

```diff
@@ -46,27 +46,22 @@ solve!(integrator)
 @test t == integrator.sol.t
 
 #callback test
-f(t,u) = 2.0*t-2.0
+g(t,u) = 2.0*t-2.0
 u0=0.0
 tspan = (0.0,4.0)
-prob = ODEProblem(f,u0,tspan)
-function condition(t,u,integrator) # Event when event_f(t,u) == 0
-  u
-end
-function affect!(integrator)
-  integrator.u = 0.7
-end
-cb = ContinuousCallback(condition,affect!)
+prob = ODEProblem(g,u0,tspan)
+saved_values = SavedValues(Float64, Float64)
+cb = SavingCallback((t,u,integrator)->u, saved_values)
 integrator = init(prob,Tsit5();dt=1//2^(4),callback = cb)
 dt = integrator.dt
 solve!(integrator)
 
-u = copy(integrator.sol.u)
-t = copy(integrator.sol.t)
-
+u = saved_values.saveval
+t = saved_values.t
+saved_values = SavedValues(Float64, Float64)
```

Comment:
@ChrisRackauckas when I remove this line 61in the `reinit_test.jl` the tests pass. But when this line is present the tests fail. Does the `reinit()` function not calling the callback ?

Reply:
You shouldn't `saved_values = SavedValues(Float64, Float64)`. That changes the pointer to the `saved_values` object. Instead, do `resize!(saved_values.t,0)`, etc.

Reply:
Thanks a lot. I have done the necessary changes. Now it's working fine.